app/models/review.rb

```diff
@@ -0,0 +1,29 @@
+class Review < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :film
+  has_many :comments, as: :commentable
+  has_many :ratings, as: :rateable
+
+  validates :review_title, :review_content, :user_id, :film_id, { presence: true }
+
+  def total_ratings
+    if ratings.any?
+      ratings.sum(:rating)
+    end
+  end
+
+  def average_rating
+    if ratings.any?
+      total_ratings/ratings.count
+    end
+  end
+
+  def positive_ratings
+    if ratings.any?
```

Comment:
Just the query will work here - `ratings.where('rating  > ?', 0).count` will give zero if there aren't any.